Classes/Helpers/OpencastHelper.php

```diff
@@ -90,11 +127,66 @@ public function getPreviewImage(File $file)
      * @return array with metadata
      */
     public function getMetaData(File $file)
+    {
+        $mediaId = $this->getOnlineMediaId($file);
+        $metadata = $this->fetchMetaData($mediaId);
+
+        return $metadata;
+    }
+
+    protected function getTitle($mediaId): string
+    {
+        return $this->fetchMetaData($mediaId)['title'];
+    }
+
+    protected function fetchMetaData($mediaId): array
     {
         $metadata = [];
 
-        $metadata['title'] = $file->getProperty('title');
+        if ($data = $this->fetchJson($mediaId)) {
+            $metadata['title'] = $data['dcTitle'];
+            $metadata['creator'] = $data['dcCreator'];
+            $metadata['publisher'] = $data['dcPublisher'];
+            $metadata['content_creation_date'] = strtotime($data['dcCreated']);
+            $metadata['content_modification_date'] = strtotime($data['modified']);
+            $metadata['keywords'] = $data['keywords'];
+            if ($data['mediapackage']) {
+                $metadata['duration'] = $data['mediapackage']['duration'] ?? 0;
+            }
+        } else {
+            // Fallback: most basic information we've got!
+            $metadata['title'] = $mediaId;
+        }
 
         return $metadata;
     }
+
+    protected function getAttachments($mediaId): ?array
+    {
+        if ($data = $this->fetchJson($mediaId)) {
+            if (is_array($data['mediapackage']) &&
+                is_array($data['mediapackage']['attachments']) &&
```

Comment:
this will give out warnings/notices, if there is no `$data['mediapackage']['attachments']` for example (`PHP Notice:  Undefined index: attachments`). 

Use `isset` instead, and skip testing every level in between:

```
            if (isset($data['mediapackage']['attachments']['attachment']) &&
                is_array($data['mediapackage']['attachments']['attachment'])) {
                ...
            }
```